src/Common/src/Interop/Windows/Winsock/Interop.select.cs

```diff
@@ -12,17 +12,17 @@ internal static partial class Winsock
         [DllImport(Interop.Libraries.Ws2_32, SetLastError = true)]
         internal static extern int select(
             [In] int ignoredParameter,
-            [In, Out] IntPtr[] readfds,
-            [In, Out] IntPtr[] writefds,
-            [In, Out] IntPtr[] exceptfds,
+            [In] ref IntPtr readfds,
+            [In] ref IntPtr writefds,
+            [In] ref IntPtr exceptfds,
             [In] ref TimeValue timeout);
 
         [DllImport(Interop.Libraries.Ws2_32, SetLastError = true)]
         internal static extern int select(
             [In] int ignoredParameter,
-            [In, Out] IntPtr[] readfds,
-            [In, Out] IntPtr[] writefds,
-            [In, Out] IntPtr[] exceptfds,
+            [In] ref IntPtr readfds,
+            [In] ref IntPtr writefds,
+            [In] ref IntPtr exceptfds,
```

Comment:
I think it'd be better to use `IntPtr*` for all of these.  Not using unsafe doesn't really help anything, as marshaling code is inherently "unsafe" and just doesn't require the `unsafe` keyword.  I see two benefits here to using `IntPtr*`: 1) When I see `ref`, I generally think it's meant to refer to a single value (even though that's obviously not required), but more importantly, 2) `ref` ends up requiring a P/Invoke marshaling frame whereas pointers are blittable, and since the rest of the arguments here are blittable as well, this won't require such a frame.

Reply:
consider it pointerified

Reply:
> 2) ref ends up requiring a P/Invoke marshaling frame whereas pointers are blittable, and since the rest of the arguments here are blittable as well, this won't require such a frame.

Reply:
>  2) ref ends up requiring a P/Invoke marshaling frame whereas pointers are blittable, and since the rest of the arguments here are blittable as well, this won't require such a frame.

When you say marshalling frame, it is "just" the infrastructure for fixing the pointer, and passing the address, though if the type behind the ref is blittable, the address will be passed without marshalling back and forth a value behind. But doing a manual fixed just before should result in the similar codegen than direct by ref arg.